groups/validate.go

```diff
@@ -104,196 +134,232 @@ func TestMergedGroupsConfig(t *testing.T) {
 	}
 
 	if !containsMergedConfig {
-		t.Errorf("Final GroupsConfig does not have merged configs from all groups.yaml files")
+		fmt.Printf("Final GroupsConfig does not have merged configs from all groups.yaml files")
+		os.Exit(1)
 	}
 	if n := len(dups); n > 0 {
-		t.Errorf("%d duplicate groups: %s", n, strings.Join(dups.List(), ", "))
+		fmt.Printf("%d duplicate groups: %s", n, strings.Join(dups.List(), ", "))
+		os.Exit(1)
 	}
 }
 
-// TestStagingEmailLength tests that the number of characters in the
+// VerifyStagingEmailLength tests that the number of characters in the
 // project name in emails used for staging repos does not exceed 18.
 //
 // This validation is needed because gcloud allows PROJECT_IDs of length
 // between 6 and 30. So after discounting the "k8s-staging" prefix,
 // we are left with 18 chars for the project name.
-func TestStagingEmailLength(t *testing.T) {
+func VerifyStagingEmailLength(config Config) {
+	primaryDomain := config.Domains[0]
 	var errs []error
-	for _, g := range cfg.Groups {
+	for _, g := range groupsConfig.Groups {
 		if strings.HasPrefix(g.EmailId, "k8s-infra-staging-") {
-			projectName := strings.TrimSuffix(strings.TrimPrefix(g.EmailId, "k8s-infra-staging-"), "@kubernetes.io")
+			projectName := strings.TrimSuffix(strings.TrimPrefix(g.EmailId, "k8s-infra-staging-"), "@"+primaryDomain)
 
 			len := utf8.RuneCountInString(projectName)
 			if len > 18 {
-				errs = append(errs, fmt.Errorf("Number of characters in project name \"%s\" should not exceed 18; is: %d", projectName, len))
+				errs = append(errs, fmt.Errorf("number of characters in project name \"%s\" should not exceed 18; is: %d", projectName, len))
```

Comment:
It seems like there's an additional assumption here about the `k8s-staging-` prefix being 12 letters.

(See the comment at the beginning of this method.) I'm fine with leaving this as a "TODO" for now, but it seems like the string `k8s-infra-staging-` should be configurable, and the `18` value should be something like `36 - len(bot-prefix)`.